src/main/java/org/apache/sysds/common/Opcodes.java

```diff
@@ -0,0 +1,328 @@
+    BATCH_NORM2D("batch_norm2d", CPType.Dnn),
+    BATCH_NORM2D_BACKWARD("batch_norm2d_backward", CPType.Dnn),
+    LSTM("lstm", CPType.Dnn),
+    LSTM_BACKWARD("lstm_backward", CPType.Dnn),
+
+    //Quaternary instruction opcodes
+    WSLOSS("wsloss", CPType.Quaternary),
+    WSIGMOID("wsigmoid", CPType.Quaternary),
+    WDIVMM("wdivmm", CPType.Quaternary),
+    WCEMM("wcemm", CPType.Quaternary),
+    WUMM("wumm", CPType.Quaternary),
+
+    //User-defined function Opcodes
+    FCALL(FunctionOp.OPCODE, CPType.FCall),
+
+    APPEND(Append.OPCODE, CPType.Append),
+    REMOVE("remove", CPType.Append),
+
+    //data generation opcodes
+    RANDOM(DataGen.RAND_OPCODE, CPType.Rand),
+    SEQUENCE(DataGen.SEQ_OPCODE, CPType.Rand),
+    STRINGINIT(DataGen.SINIT_OPCODE, CPType.StringInit),
+    SAMPLE(DataGen.SAMPLE_OPCODE, CPType.Rand),
+    TIME(DataGen.TIME_OPCODE, CPType.Rand),
+    FRAME(DataGen.FRAME_OPCODE, CPType.Rand),
+
+    CTABLE("ctable", CPType.Ctable),
+    CTABLEEXPAND("ctableexpand", CPType.Ctable),
+
+    //central moment, covariance, quantiles (sort/pick)
+    CM("cm", CPType.CentralMoment),
+    COV("cov", CPType.Covariance),
+    QSORT("qsort", CPType.QSort),
+    QPICK("qpick", CPType.QPick),
+
+    RIGHT_INDEX(RightIndex.OPCODE, CPType.MatrixIndexing),
+    LEFT_INDEX(LeftIndex.OPCODE, CPType.MatrixIndexing),
+
+    TSMM("tsmm", CPType.MMTSJ),
+    PMM("pmm", CPType.PMMJ),
+    MMCHAIN("mmchain", CPType.MMChain),
+
+    QR("qr", CPType.MultiReturnBuiltin),
+    LU("lu", CPType.MultiReturnBuiltin),
+    EIGEN("eigen", CPType.MultiReturnBuiltin),
+    FFT("fft", CPType.MultiReturnBuiltin),
+    IFFT("ifft", CPType.MultiReturnComplexMatrixBuiltin),
+    FFT_LINEARIZED("fft_linearized", CPType.MultiReturnBuiltin),
+    IFFT_LINEARIZED("ifft_linearized", CPType.MultiReturnComplexMatrixBuiltin),
+    STFT("stft", CPType.MultiReturnComplexMatrixBuiltin),
+    SVD("svd", CPType.MultiReturnBuiltin),
+    RCM("rcm", CPType.MultiReturnComplexMatrixBuiltin),
+
+    PARTITION("partition", CPType.Partition),
+    COMPRESS(Compression.OPCODE, CPType.Compression),
+    DECOMPRESS(DeCompression.OPCODE, CPType.DeCompression),
+    SPOOF("spoof", CPType.SpoofFused),
+    PREFETCH("prefetch", CPType.Prefetch),
+    EVICT("_evict", CPType.EvictLineageCache),
+    BROADCAST("broadcast", CPType.Broadcast),
+    TRIGREMOTE("trigremote", CPType.TrigRemote),
+    LOCAL(Local.OPCODE, CPType.Local),
+
+    SQL("sql", CPType.Sql);
+
+
+    // Constructor
+    Opcodes(String name, CPType type) {
+        this._name = name;
+        this._type = type;
+    }
+
+    // Fields
+    private final String _name;
+    private final CPType _type;
+
+    private static final Map<String, Opcodes> _lookupMap = new HashMap<>();
+
+    // Initialize lookup map
+    static {
+        for (Opcodes op : EnumSet.allOf(Opcodes.class)) {
+            _lookupMap.put(op.getName(), op);
+        }
+    }
+
+    // Getters
+    public String getName() {
+        return _name;
+    }
```

Comment:
instead of this method, please overwrite `toString` which then allows to use `OpCode.XXX` instead of `OpCode.XXX.getName()`